posts/map_blocks_example.md

```diff
@@ -0,0 +1,191 @@
+
+## Imports and Data Ingestion
+
+We use the following libraries/packages within this example
+
+```python
+import os
+
+from scipy import interpolate
+import numpy as np
+import xarray as xr
+
+import matplotlib.pyplot as plt
+
+import pop_tools
+```
+
+Next, we read in the data using `xarray` and subset for temperature and salinity, subsetting for a smaller portion of the domain for testing purposes
+
+```
+ds = xr.merge((
+    xr.open_dataset('/glade/p/cgd/oce/projects/cesm2-marbl/xpersist_cache/3d_fields/TEMP-presentday-monclim.nc'),
+    xr.open_dataset('/glade/p/cgd/oce/projects/cesm2-marbl/xpersist_cache/3d_fields/SALT-presentday-monclim.nc'),
+))
+
+
+ds = ds[['TEMP', 'SALT']]
+
+# subset for testing purposes
+ds = ds.isel(member_id=slice(0, 2), nlat=slice(2, 5), nlon=slice(1, 5))
+```
+
+## Data Operation
+
+Next, we setup a function to compute the mixed layer depth (MLD). The MLD is defined as the point in the water column where the potential density (`sigma`) exceeds the surface density by a specified threshold (`dsigma`).
+
+```python
+def mld_dsigma(SALT, TEMP, dsigma=0.03, rho_chunks={'nlat': 16, 'nlon': 16}):
+    """
+    Compute MLD based on ∆σ criterion. Uses xarray.map_blocks.
+
+    Parameters
+    ----------
+
+    SALT : xarray.DataArray
+      Salinity
+    TEMP : xarray.DataArray
+      Potential temperature
+    dsigma : float, optional
+      The value for ∆σ.
+    rho_chunks: dictionary, optional
+      Dimension chunk parameters to chunk rho by
+
+    Returns
+    -------
+
+    MLD : xarray.DataArray
+      The MLD (m) defined as the point in the water column where
+      density exceeds rho[0] + dsigma.
+    """
+
+    # determine dimensionality
+    dims_in = SALT.dims
+    assert dims_in == TEMP.dims, 'dimension mismatch'
+    assert 'z_t' in SALT.coords, 'z_t not found in SALT coords'
+
+    # drop ancillary coordinates (this may not be necessary)
+    SALT = SALT.reset_coords(drop=True)
+    TEMP = TEMP.reset_coords(drop=True)
+
+    # compute density
+    rho = pop_tools.eos(SALT.chunk({'z_t': 10}),
+                        TEMP.chunk({'z_t': 10}),
+                        depth=SALT.z_t * 0.).compute()
+
+    # add these coordinates which creep in somewhere, I think when xarray does the unstack
+    # without these, I get an error: not expecting {'nlat', 'nlon'}
+    # chunking here is arbitrary and maybe suitable for the global POP_gx1v7 grid
+    if 'nlat' in rho.dims and 'nlon' in rho.dims:
+        rho = rho.assign_coords({
+            'nlat': xr.DataArray(np.arange(len(SALT.nlat)), dims=('nlat')),
+            'nlon': xr.DataArray(np.arange(len(SALT.nlon)), dims=('nlon')),
+        })
+
+    rho = rho.chunk(rho_chunks).persist()
+
+    # Setup a template
```

Comment:
We should explain why this is needed. There is some text here: https://xarray.pydata.org/en/stable/dask.html#map-blocks . We should also link to that doc page.